pkg/store/container/metadata_test.go

```diff
@@ -37,17 +37,35 @@ func TestMetadataEncodeDecode(t *testing.T) {
 		},
 		ImageRef: "test-image-ref",
 	}
+
 	assert := assertlib.New(t)
-	data, err := meta.Encode()
-	assert.NoError(err)
 	newMeta := &Metadata{}
-	assert.NoError(newMeta.Decode(data))
+	newVerMeta := &versionedMetadata{}
+
+	data, err := json.Marshal(meta)
+	assert.NoError(err)
+	assert.NoError(json.Unmarshal(data, newMeta))
+	assert.Equal(meta, newMeta)
```

Comment:
We don't know whether `MarshalJSON` or `UnmarshalJSON` is actually called during this.
The test is a little bit confusing.

How about this?
1) Your first case. (verify that data could be stored and recovered)
2) Make sure `json.Marshal(Metadata)` and `json.Marshal(versionedMetadata)` get the same bytes result; (verify that data is stored as versioned data, case1) has proved that versioned data could be recovered)
3) `json.Marshal(unsupportedVersionedMetadata)` and then `json.Unmarshal(data)` should get an error; (verify that unsupported versioned data won't be unmarshalled)

We may want to add some `t.Logf` to make the test more clear.

Reply:
done